lisa/operating_system.py

```diff
@@ -198,19 +307,57 @@ def name_pattern(cls) -> Pattern[str]:
     def _initialize_package_installation(self) -> None:
         self._node.execute("apt-get update", sudo=True)
 
-    def _install_packages(self, packages: Union[List[str]]) -> None:
+    def _install_packages(
+        self, packages: Union[List[str]], signed: bool = True
+    ) -> None:
         command = (
             f"DEBIAN_FRONTEND=noninteractive "
             f"apt-get -y install {' '.join(packages)}"
         )
-        self._node.execute(command, sudo=True)
+        if not signed:
+            command = command.__add__(" --allow-unauthenticated")
+
+        install_result = self._node.execute(command, sudo=True)
+        if install_result.exit_code != 0:
+            raise LisaException(
+                f"Failed to install {' '.join(packages)}."
+                f" stdout: {install_result.stdout}"
+            )
+
+    def _query_package(self, package: str, signed: bool = True) -> bool:
+        command = f"apt list --installed | grep -Ei {package}"
+        result = self._node.execute(command, sudo=True)
+        if result.exit_code == 0:
+            for row in result.stdout.split("\n"):
+                if package in row:
+                    return True
+        return False
+
+    def _get_os_version(self) -> OsVersion:
+        cmd_result = self._node.execute(cmd="lsb_release -as", no_error_log=True)
+        if cmd_result.exit_code == 0 and cmd_result.stdout != "":
+            for row in cmd_result.stdout.split("\n"):
+                if ": " in row:
```

Comment:
Use regular expressions will be more clear.

Reply:
fixed, it's just searching though.

Reply:
refer to `lisa\tools\lsvmbus.py`, the name captured regexp can make the logic more clear.

Reply:
Added Regex